[ecoflow] Losing the internet connection sometimes breaks MQTT OpenHAB wide (blocks threads)

## Expected Behavior
If the internet connection goes down it should not have a negative impact on mqtt in OpenHAB in general and shouldn't break other (local) mqtt connections.

## Current Behavior

When the configured ecoflow things are online and the internet connection goes down for a few seconds it sometimes (not every time) happens that some mqtt items (and ecoflow items) don't receive updates anymore until OpenHAB is restarted.

### When the problem occurs I've noticed that:

1. The OpenHAB console command ```threads --monitors --locks | grep BLOCKED``` shows blocked threads and returns lines like ```RxComputationThreadPool-1" Id=699 in BLOCKED on lock=java.lang.Object@4faca7dd```. Those don't disappear until I restart OpenHAB. I've tried only restarting some mqtt/ecoflow/hivemq/rxjava bundles, but without success. Restarting the mqtt bundle or disabling/enabling some mqtt broker things then also leads to ```[WARN ] [qtt.generic.AbstractMQTTThingHandler] - unsubscription on disposal failed for mqtt:topic:``` messages in the log.
2. After some time (usually hours) ```[WARN ] [.incoming.MqttIncomingPublishService] - QoS 0 publish message dropped``` messages flood the log.

### With debug logging enabled for the binding:

1. I always get ```[DEBUG] [w.internal.handler.EcoflowApiHandler] - MQTT disconnected (source CLIENT): java.net.SocketException: Connection reset``` when the internet connection goes down.
2. Usually before I get the blocked threads and mqtt fails, something like this is shown in the log (but also the threads are not _always_ getting blocked when this stack trace appears; serial numbers replaced with ***):
```
2026-07-18 18:05:45.053 [DEBUG] [w.internal.handler.EcoflowApiHandler] - MQTT disconnected (source CLIENT): java.net.SocketException: Connection reset
2026-07-18 18:06:00.055 [DEBUG] [w.internal.handler.EcoflowApiHandler] - Could not establish MQTT connection
org.openhab.binding.ecoflow.internal.api.EcoflowApiException: java.util.concurrent.TimeoutException: Total timeout 10000 ms elapsed
	at org.openhab.binding.ecoflow.internal.api.EcoflowApi.executeRequest(EcoflowApi.java:197)
	at org.openhab.binding.ecoflow.internal.api.EcoflowApi.createMqttLogin(EcoflowApi.java:116)
	at org.openhab.binding.ecoflow.internal.handler.EcoflowApiHandler.establishMqttConnection(EcoflowApiHandler.java:221)
	at org.openhab.binding.ecoflow.internal.util.SchedulerTask.run(SchedulerTask.java:95)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.util.concurrent.TimeoutException: Total timeout 10000 ms elapsed
	at org.eclipse.jetty.client.HttpConnection$RequestTimeouts.onExpired(HttpConnection.java:334)
	at org.eclipse.jetty.client.HttpConnection$RequestTimeouts.onExpired(HttpConnection.java:314)
	at org.eclipse.jetty.io.CyclicTimeouts.onTimeoutExpired(CyclicTimeouts.java:110)
	at org.eclipse.jetty.io.CyclicTimeouts.access$100(CyclicTimeouts.java:50)
	at org.eclipse.jetty.io.CyclicTimeouts$Timeouts.onTimeoutExpired(CyclicTimeouts.java:196)
	at org.eclipse.jetty.io.CyclicTimeout$Wakeup.run(CyclicTimeout.java:298)
	... 6 more
2026-07-18 18:06:05.187 [DEBUG] [w.internal.handler.EcoflowApiHandler] - Opening MQTT connection
2026-07-18 18:06:05.303 [DEBUG] [w.internal.handler.EcoflowApiHandler] - Established MQTT connection
2026-07-18 18:06:05.304 [DEBUG] [w.internal.handler.EcoflowApiHandler] - Subscribing for updates from ***
2026-07-18 18:06:05.472 [DEBUG] [w.internal.handler.EcoflowApiHandler] - Subscribing for updates from ***
2026-07-18 18:06:05.659 [DEBUG] [w.internal.handler.EcoflowApiHandler] - Subscribing for updates from ***
2026-07-18 18:07:05.657 [DEBUG] [w.internal.handler.EcoflowApiHandler] - MQTT disconnected (source CLIENT): java.net.SocketException: Connection reset
```
3. I've also noticed the following stack trace, but this seems less related to the threads actually getting blocked (IPs replaced with ***):
```
[DEBUG] [w.internal.handler.EcoflowApiHandler] - Could not establish MQTT connection
org.openhab.binding.ecoflow.internal.api.EcoflowApiException: java.util.concurrent.ExecutionException: java.io.EOFException: @697fa213::DecryptedEndPoint@93534c9{l=/***:51074,r=api.ecoflow.com/***:443,OPEN,fill=-,flush=-,to=68111/0}
	at org.openhab.binding.ecoflow.internal.api.EcoflowApi.executeRequest(EcoflowApi.java:197)
	at org.openhab.binding.ecoflow.internal.api.EcoflowApi.createMqttLogin(EcoflowApi.java:116)
	at org.openhab.binding.ecoflow.internal.handler.EcoflowApiHandler.establishMqttConnection(EcoflowApiHandler.java:221)
	at org.openhab.binding.ecoflow.internal.util.SchedulerTask.run(SchedulerTask.java:95)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.util.concurrent.ExecutionException: java.io.EOFException: @697fa213::DecryptedEndPoint@93534c9{l=/***:51074,r=api.ecoflow.com/***:443,OPEN,fill=-,flush=-,to=68111/0}
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:732)
	at org.openhab.binding.ecoflow.internal.api.EcoflowApi.executeRequest(EcoflowApi.java:191)
	... 9 more
Caused by: java.io.EOFException: @697fa213::DecryptedEndPoint@93534c9{l=/***:51074,r=api.ecoflow.com/***:443,OPEN,fill=-,flush=-,to=68111/0}
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.earlyEOF(HttpReceiverOverHTTP.java:385)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1621)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.shutdown(HttpReceiverOverHTTP.java:269)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:185)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:80)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:131)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:172)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:555)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:410)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:164)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	... 1 more
```

## Steps to Reproduce (for Bugs)
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
1. Have the ecoflow binding/things enabled and some ecoflow devices configured and online. The bug only occurs when the ecoflow devices are actually online and sending updates to the ecoflow mqtt server and the plugin subscribes to those topics.
2. Disconnect and connect the internet a few times. I've done this by stopping and starting the WAN interface on my OpenWRT router using this command: ```for i in $(seq 1 50); do echo "Down"; ifdown wan_interface; sleep 5; echo "Up"; ifup wan_interface; sleep 10; done``` where "wan_interface" is the wan interface used. This pretty reliably resulted in the blocked threads and some mqtt items and the ecoflow items not getting updated anymore.
3. Check for blocked "**RxComputationThreadPool**" threads using this OpenHAB console command: ```threads --monitors --locks | grep BLOCKED```. If you see blocked threads, some mqtt items should now not get updated anymore (it seems how much items are affected depends on how many threads are blocked). After a few hours this message should start flooding the log: ```[WARN ] [.incoming.MqttIncomingPublishService] - QoS 0 publish message dropped```

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->
Recently we switched to an internet provider that does a connection reset each day. After the forced daily disconnect some mqtt items (and the ecoflow items) often didn't receive updates anymore until I restarted OpenHAB, even though the mqtt items and the mqtt brokers are completely local. Sending commands to them from OpenHAB still worked though. The devices were still connected to mosquitto and I could see them sending new states to the broker. OpenHAB just didn't process them anymore.
I'm pretty confident that this is caused by the ecoflow plugin, as I could only reproduce this when the ecoflow account thing was enabled and only when the configured ecoflow devices/things were online. To my knowledge the ecoflow binding is also the only add-on I use that is connecting to an mqtt broker on the internet.
I've currently added a JS rule that detects the blocked threads and shuts down OpenHAB, so the container gets restarted (by systemd in my case).

## Your Environment
* Version used: OpenHAB 5.2.0, Binding 5.2.0
* Server: Debian Trixie with OpenHAB running as a Podman container (official Docker Hub image).

--- a/bundles/org.openhab.binding.ecoflow/src/main/java/org/openhab/binding/ecoflow/internal/handler/EcoflowApiHandler.java
+++ b/bundles/org.openhab.binding.ecoflow/src/main/java/org/openhab/binding/ecoflow/internal/handler/EcoflowApiHandler.java
@@ -140,6 +140,8 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
                 if (mqttConnection != null) {
                     subscribeTask.cancel();
                     subscribeTask.submit();
+                } else {
+                    mqttConnectTask.submit();
                 }
             }
         }

--- a/bundles/org.openhab.binding.ecoflow/src/main/java/org/openhab/binding/ecoflow/internal/handler/EcoflowApiHandler.java
+++ b/bundles/org.openhab.binding.ecoflow/src/main/java/org/openhab/binding/ecoflow/internal/handler/EcoflowApiHandler.java
@@ -20,6 +20,8 @@ import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.function.Consumer;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
@@ -68,18 +70,32 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
     private Optional<EcoflowDeviceDiscoveryService> discoveryService = Optional.empty();
     private final SchedulerTask initTask;
     private final SchedulerTask mqttConnectTask;
+    private final SchedulerTask subscribeTask;
     private final HttpClient httpClient;
     private @Nullable EcoflowApi api;
-    private @Nullable MqttConnection mqttConnection;
 
     private final Object mqttConnectionLock = new Object();
-    private final Map<String, AbstractEcoflowHandler> activeChildHandlers = new HashMap<>();
+    private @Nullable MqttConnection mqttConnection;
+
+    private class HandlerData {
+        private final AbstractEcoflowHandler handler;
+        private final String serialNumber;
+        private boolean subscribed = false;
+
+        public HandlerData(AbstractEcoflowHandler handler) {
+            this.handler = handler;
+            this.serialNumber = handler.getSerialNumber();
+        }
+    }
+
+    private final Map<String, HandlerData> activeChildHandlers = new HashMap<>();
 
     public EcoflowApiHandler(Bridge bridge, HttpClient httpClient) {
         super(bridge);
         this.httpClient = httpClient;
         this.initTask = new SchedulerTask(scheduler, logger, "API Init", this::initApi);
         this.mqttConnectTask = new SchedulerTask(scheduler, logger, "MQTT Connection", this::establishMqttConnection);
+        this.subscribeTask = new SchedulerTask(scheduler, logger, "MQTT Subscription", this::updateSubscriptions);
     }
 
     public void setDiscoveryService(EcoflowDeviceDiscoveryService discoveryService) {
@@ -117,22 +133,13 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
     @Override
     public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
         super.childHandlerInitialized(childHandler, childThing);
-        logger.debug("child handler {} initialized", childHandler);
         if (childHandler instanceof AbstractEcoflowHandler deviceHandler) {
             synchronized (mqttConnectionLock) {
-                MqttConnection connection = mqttConnection;
-                activeChildHandlers.put(deviceHandler.getSerialNumber(), deviceHandler);
-                if (connection != null) {
-                    try {
-                        subscribeForDeviceLocked(connection, deviceHandler.getSerialNumber());
-                        deviceHandler.handleMqttConnected();
-                    } catch (EcoflowApiException e) {
-                        logger.debug("{}: Could not subscribe for MQTT updates, re-scheduling connection",
-                                deviceHandler.getSerialNumber());
-                        mqttConnectTask.schedule(5);
-                    }
-                } else {
-                    mqttConnectTask.submit();
+                String serialNumber = deviceHandler.getSerialNumber();
+                activeChildHandlers.put(serialNumber, new HandlerData(deviceHandler));
+                if (mqttConnection != null) {
+                    subscribeTask.cancel();
+                    subscribeTask.submit();
                 }
             }
         }
@@ -141,15 +148,19 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
     @Override
     public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
         super.childHandlerDisposed(childHandler, childThing);
-        logger.debug("child handler {} disposed", childHandler);
         if (childHandler instanceof AbstractEcoflowHandler deviceHandler) {
+            final MqttConnection connectionToTearDown;
             synchronized (mqttConnectionLock) {
-                final MqttConnection connection = mqttConnection;
                 activeChildHandlers.remove(deviceHandler.getSerialNumber());
-                if (activeChildHandlers.isEmpty() && connection != null) {
-                    connection.disconnect();
+                if (activeChildHandlers.isEmpty()) {
+                    connectionToTearDown = mqttConnection;
                     mqttConnection = null;
+                } else {
+                    connectionToTearDown = null;
                 }
+            }
+            if (connectionToTearDown != null) {
+                connectionToTearDown.disconnect();
             }
         }
     }
@@ -205,47 +216,70 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
             return;
         }
 
+        final MqttConnection oldConnection;
         synchronized (mqttConnectionLock) {
-            MqttConnection oldConnection = mqttConnection;
-            if (oldConnection != null) {
-                try {
-                    oldConnection.disconnect().get();
-                } catch (InterruptedException | ExecutionException e) {
-                    logger.debug("Could not discard MQTT connection", e);
-                }
-            }
-
+            oldConnection = mqttConnection;
             mqttConnection = null;
-
+        }
+        if (oldConnection != null) {
             try {
-                MqttConnectionData connectData = api.createMqttLogin();
-                Mqtt3AsyncClient client = establishMqttConnection(connectData);
-
-                MqttConnection connection = new MqttConnection(client, connectData.userName);
-                for (String serialNumber : activeChildHandlers.keySet()) {
-                    subscribeForDeviceLocked(connection, serialNumber);
-                }
-
-                mqttConnection = connection;
-
-                for (AbstractEcoflowHandler handler : activeChildHandlers.values()) {
-                    handler.handleMqttConnected();
-                }
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-            } catch (EcoflowApiException e) {
-                logger.debug("Could not establish MQTT connection", e);
-                mqttConnectTask.schedule(5);
+                oldConnection.disconnect().get();
+            } catch (InterruptedException | ExecutionException e) {
+                logger.debug("Could not discard MQTT connection", e);
             }
+        }
+
+        try {
+            MqttConnectionData connectData = api.createMqttLogin();
+            Mqtt3AsyncClient client = establishMqttConnection(connectData);
+
+            MqttConnection connection = new MqttConnection(client, connectData.userName);
+            synchronized (mqttConnectionLock) {
+                subscribeTask.cancel();
+                activeChildHandlers.values().forEach(data -> data.subscribed = false);
+                mqttConnection = connection;
+                subscribeTask.submit();
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (EcoflowApiException e) {
+            logger.debug("Could not establish MQTT connection", e);
+            mqttConnectTask.schedule(5);
         }
     }
 
-    private void subscribeForDeviceLocked(MqttConnection connection, String serialNumber) throws EcoflowApiException {
-        try {
-            logger.debug("Subscribing for updates from {}", serialNumber);
-            connection.subscribeForDevice(serialNumber, this::handleQuotaMessage, this::handleStatusMessage);
-        } catch (ExecutionException | InterruptedException e) {
-            throw new EcoflowApiException(e);
+    private void updateSubscriptions() {
+        final MqttConnection connection;
+        final List<HandlerData> handlersToBeSubscribed;
+
+        synchronized (mqttConnectionLock) {
+            connection = mqttConnection;
+            handlersToBeSubscribed = activeChildHandlers.values().stream().filter(data -> !data.subscribed).toList();
+        }
+        if (connection == null) {
+            logger.trace("MQTT not connected, postponing subscription");
+            return;
+        }
+
+        for (HandlerData data : handlersToBeSubscribed) {
+            try {
+                logger.debug("Subscribing for updates from {}", data.serialNumber);
+                connection.subscribeForDevice(data.serialNumber, this::handleQuotaMessage, this::handleStatusMessage)
+                        .get(2, TimeUnit.SECONDS);
+                synchronized (mqttConnectionLock) {
+                    data.subscribed = true;
+                    data.handler.handleMqttConnected();
+                }
+            } catch (ExecutionException | InterruptedException | TimeoutException e) {
+                logger.debug("Could not subscribe for MQTT updates for device {}:", data.serialNumber, e);
+            }
+        }
+
+        synchronized (mqttConnectionLock) {
+            if (activeChildHandlers.values().stream().anyMatch(data -> !data.subscribed)) {
+                logger.debug("Not all devices subscribed yet, retrying");
+                subscribeTask.submit();
+            }
         }
     }
 
@@ -258,7 +292,10 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
         final MqttClientDisconnectedListener disconnectListener = ctx -> {
             boolean expectedShutdown = ctx.getSource() == MqttDisconnectSource.USER
                     && ctx.getCause() instanceof Mqtt3DisconnectException;
-            mqttConnection = null;
+            synchronized (mqttConnectionLock) {
+                mqttConnection = null;
+                subscribeTask.cancel();
+            }
             if (!expectedShutdown) {
                 logger.debug("MQTT disconnected (source {}): {}", ctx.getSource(), ctx.getCause().getMessage());
                 mqttConnectTask.schedule(5);
@@ -312,7 +349,8 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
             throw new IllegalStateException("Unexpected topic " + topic);
         }
         synchronized (mqttConnectionLock) {
-            return activeChildHandlers.get(levels.get(3));
+            var data = activeChildHandlers.get(levels.get(3));
+            return data != null ? data.handler : null;
         }
     }
 
@@ -329,11 +367,14 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
             topicBase = String.format("/open/%s/", userName);
         }
 
-        void subscribeForDevice(String serialNumber, Consumer<@Nullable Mqtt3Publish> quotaHandler,
+        CompletableFuture<Void> subscribeForDevice(String serialNumber, Consumer<@Nullable Mqtt3Publish> quotaHandler,
                 Consumer<@Nullable Mqtt3Publish> statusHandler) throws ExecutionException, InterruptedException {
             String deviceTopicBase = topicBase + serialNumber + "/";
-            client.subscribeWith().topicFilter(deviceTopicBase + "quota").callback(quotaHandler).send().get();
-            client.subscribeWith().topicFilter(deviceTopicBase + "status").callback(statusHandler).send().get();
+            var quotaSubFuture = client.subscribeWith().topicFilter(deviceTopicBase + "quota").callback(quotaHandler)
+                    .send();
+            var statusSubFuture = client.subscribeWith().topicFilter(deviceTopicBase + "status").callback(statusHandler)
+                    .send();
+            return CompletableFuture.allOf(quotaSubFuture, statusSubFuture);
         }
 
         CompletableFuture<Void> disconnect() {

--- a/bundles/org.openhab.binding.ecoflow/src/main/java/org/openhab/binding/ecoflow/internal/handler/EcoflowApiHandler.java
+++ b/bundles/org.openhab.binding.ecoflow/src/main/java/org/openhab/binding/ecoflow/internal/handler/EcoflowApiHandler.java
@@ -226,7 +226,9 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
         if (oldConnection != null) {
             try {
                 oldConnection.disconnect().get();
-            } catch (InterruptedException | ExecutionException e) {
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } catch (ExecutionException e) {
                 logger.debug("Could not discard MQTT connection", e);
             }
         }
@@ -272,7 +274,9 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
                     data.subscribed = true;
                     data.handler.handleMqttConnected();
                 }
-            } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } catch (ExecutionException | TimeoutException e) {
                 logger.debug("Could not subscribe for MQTT updates for device {}:", data.serialNumber, e);
             }
         }
@@ -370,7 +374,7 @@ public class EcoflowApiHandler extends BaseBridgeHandler {
         }
 
         CompletableFuture<Void> subscribeForDevice(String serialNumber, Consumer<@Nullable Mqtt3Publish> quotaHandler,
-                Consumer<@Nullable Mqtt3Publish> statusHandler) throws ExecutionException, InterruptedException {
+                Consumer<@Nullable Mqtt3Publish> statusHandler) {
             String deviceTopicBase = topicBase + serialNumber + "/";
             var quotaSubFuture = client.subscribeWith().topicFilter(deviceTopicBase + "quota").callback(quotaHandler)
                     .send();

